// sockjs/lib/chunking-test.js
(function() {
  var utils;
  var __bind = function(fn, me){ return function(){ return fn.apply(me, arguments); }; };
  utils = require('./utils');
  exports.app = {
    chunking_test: function(req, res, _, next_filter) {
      var write;
      res.setHeader('Content-Type', 'application/javascript; charset=UTF-8');
      res.writeHead(200);
      write = __bind(function(payload) {
        try {
          return res.write(payload + '\n');
        } catch (x) {

        }
      }, this);
      utils.timeout_chain([
        [
          0, __bind(function() {
            return write('h');
          }, this)
        ], [
          1, __bind(function() {
            return write(Array(2049).join(' ') + 'h');
          }, this)
        ], [
          5, __bind(function() {
            return write('h');
          }, this)
        ], [
          25, __bind(function() {
            return write('h');
          }, this)
        ], [
          125, __bind(function() {
            return write('h');
          }, this)
        ], [
          625, __bind(function() {
            return write('h');
          }, this)
        ], [
          3125, __bind(function() {
            write('h');
            return res.end();
          }, this)
        ]
      ]);
      return true;
    },
    info: function(req, res, _) {
      var info;
      info = {
        websocket: this.options.websocket,
        origins: this.options.origins,
        cookie_needed: !!this.options.jsessionid,
        entropy: utils.random32()
      };
      res.setHeader('Content-Type', 'application/json; charset=UTF-8');
      res.writeHead(200);
      return res.end(JSON.stringify(info));
    },
    info_options: function(req, res) {
      res.statusCode = 204;
      res.setHeader('Allow-Control-Allow-Methods', 'OPTIONS, GET');
      res.setHeader('Access-Control-Max-Age', res.cache_for);
      return '';
    }
  };
}).call(this);
